### PR TITLE
FIX: App store apis merge into single and package refactor

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

Api Modification and Package Refactor for Deployment History, Deployment History Info, Rollback.

Combine rest handlers for same purpose of apis like deployment history will have single api for both hyperion and full mode.
Packages refactor

Fixes # (issue) : https://github.com/devtron-labs/devtron/issues/1419

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Deployment History for both EA_Only/FULL
- [x] Deployment History Info  for both EA_Only/FULL
- [x] Rollback for both EA_Only/FULL

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [x] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

File: specs/api-spec.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/RollbackReleaseResponse"
-  /orchestrator/application/rollback:
+  /orchestrator/application/rollback/deprecated:
```
Comment (manish-agrawal-ai) [inline specs/api-spec.yaml:22]: deprecated?

File: api/router/router.go [hunk 2]
```diff
@@ -330,4 +331,10 @@ func (r MuxRouter) Init() {
 	deploymentConfigSubRouter := r.Router.PathPrefix("/orchestrator/deployment/template").Subrouter()
 	r.deploymentConfigRouter.Init(deploymentConfigSubRouter)
 	// deployment router ends
+
+	//GitOps,Acd + HelmCLi both apps deployment related api's
+	helmApp := r.Router.PathPrefix("/orchestrator/application").Subrouter()
```
Comment (manish-agrawal-ai) [inline api/router/router.go:336]: rename helmApp to applicationSubRouter

File: cmd/external-app/router.go [hunk 3]
```diff
@@ -126,6 +129,8 @@ func (r *MuxRouter) Init() {
 
 	helmApp := r.Router.PathPrefix("/orchestrator/application").Subrouter()
```
Comment (manish-agrawal-ai) [inline cmd/external-app/router.go:130]: rename helmApp to applicationSubRouter

**Review by manish-agrawal-ai — COMMENTED** [review]
check comment

File: cmd/external-app/router.go (hunk 3 above)
Comment (vikramdevtron) [reply cmd/external-app/router.go:130]: done

File: api/router/router.go (hunk 2 above)
Comment (vikramdevtron) [reply api/router/router.go:336]: done

File: specs/api-spec.yaml (hunk 1 above)
Comment (vikramdevtron) [reply specs/api-spec.yaml:22]: removing this, as two apis merged in one

**Pull request merged**